frontend/src/components/admin/assessment-creation/AssessmentEditorHeader.tsx

```diff
@@ -77,12 +74,16 @@ const AssessmentEditorHeader = ({
     onPublishModalOpen();
   };
 
-  const handleSave = handleSubmit(onSave, onError);
-  const handlePublish = handleSubmit(onPublish, onError);
-  const handleConfirmPublish = handleSubmit(onConfirmPublish, onError);
-  const handleConfirmArchive = handleSubmit(onConfirmArchive, onError);
-  const handleDelete = handleSubmit(onDelete, onError);
-  const handleCloseEditor = () => history.goBack();
+  const { getValues } = useFormContext<TestRequest>();
+
+  // We need validation on these actions.
+  const handleSave = useActionFormHandler(onSave);
+  const handlePublish = useActionFormHandler(onPublish);
+  const handleConfirmPublish = useActionFormHandler(onConfirmPublish);
```

Comment:
do we need to validate again for confirmPublish if we already did for publish?

Reply:
True